src/Swarm/Game/Step.hs

```diff
@@ -139,15 +139,14 @@ gameTick = do
       -- fresh CESK machine, using a copy of the current game state.
       v <- runThrow @Exn . evalState @GameState g $ evalPT (obj ^. objectiveCondition)
       let markWin = winCondition .= maybe (Won False) WinConditions (NE.nonEmpty objs)
-      case v of
+      case stripVResult <$> v of
         -- Log exceptions in the message queue so we can check for them in tests
         Left exn -> do
           let h = hypotheticalRobot (Out VUnit emptyStore []) 0
           em <- use entityMap
           m <- evalState @Robot h $ createLogEntry ErrorTrace (formatExn em exn)
           emitMessage m
         Right (VBool res) -> when res markWin
-        Right (VResult (VBool res) _env) -> when res markWin
```

Comment:
#927 continues to accrue merge conflicts in [this part of the code](https://github.com/swarm-game/swarm/pull/927#discussion_r1061820199).

Reply:
@kostmo if you are OK with letting me push to your branch, I can fix it for you. :wink: 

We have a historic number of open PRs now, so I will try to close mine and review and help where I can.

Afterwards, we should make a release, there is a lot of cool stuff to share. :+1: